src/hooks/twilio/useVideoRoomState.tsx

```diff
@@ -22,58 +19,57 @@ export const useVideoRoomState = (
   roomName?: string | undefined,
   activeParticipantByDefault = true
 ) => {
-  const [room, setRoom] = useState<Room>();
-  const [videoError, setVideoError] = useState("");
-  const dismissVideoError = useCallback(() => setVideoError(""), []);
-  const [localParticipant, setLocalParticipant] = useState<LocalParticipant>();
-  const [
-    participants,
-    { upsert: upsertParticipant, filter: filterParticipants },
-  ] = useList<ParticipantWithUser<RemoteParticipant>>([]);
-
-  const disconnect = useCallback(() => {
-    setRoom((currentRoom) => {
-      if (currentRoom?.localParticipant?.state !== "connected")
-        return currentRoom;
-
-      currentRoom.localParticipant.tracks.forEach((trackPublication) => {
-        (trackPublication.track as LocalVideoTrack).stop();
-      });
+  const {
+    joinChannel,
+    localParticipant,
+    remoteParticipants,
+    status,
+    disconnect,
+  } = useVideoComms();
 
-      currentRoom.disconnect();
+  const [idToUserMap, setIdToUserMap] = useState<
+    Record<string, WithId<User> | undefined>
+  >({});
 
-      return undefined;
-    });
-  }, []);
+  const [videoError, setVideoError] = useState("");
+  const dismissVideoError = useCallback(() => setVideoError(""), []);
 
   const {
-    isShown: isActiveParticipant,
-
     show: becomeActiveParticipant,
     hide: becomePassiveParticipant,
   } = useShowHide(activeParticipantByDefault);
 
-  const participantConnected = useCallback(
-    async (participant: RemoteParticipant) => {
-      const user = await getUser(participant.identity);
-      upsertParticipant(
-        (existing) => existing.participant.identity === participant.identity,
-        {
-          participant: participant,
-          user,
+  useEffect(() => {
+    return () => {
+      disconnect();
+    };
+  }, [disconnect]);
+
+  const participants = useMemo(
+    () =>
+      remoteParticipants.map((remoteParticipant) => {
+        if (!(remoteParticipant.sparkleId in idToUserMap)) {
```

Comment:
[code smell] keyword `in`. doesn't it also check the prototype?

Probably not an issue in this case, yet there should be other ways to write the same. Good enough in this case.